Apply AutoCloseableMustBeClosed

schlosna opened the pull request:
Apply `AutoCloseableMustBeClosed` from https://github.com/palantir/gradle-baseline/pull/1673

## Before this PR
Methods returning `AutoCloseable` were not marked as `@MustBeClosed`

## After this PR
<!-- User-facing outcomes this PR delivers go below -->
==COMMIT_MSG==
Methods returning `AutoCloseable` marked as `@MustBeClosed`

./gradlew clean compileJava compileTestJava -PerrorProneApply=AutoCloseableMustBeClosed
./gradlew format
==COMMIT_MSG==

## Possible downsides?
<!-- Please describe any way users could be negatively affected by this PR. -->

schlosna committed:
Add generated changelog entries

carterkozak commented on tritium-metrics/src/main/java/com/palantir/tritium/metrics/InstrumentedSslServerSocketFactory.java:
```diff
@@ -53,21 +54,25 @@
     }
 
     @Override
+    @MustBeClosed
```
Perhaps we should disable these on overrides unless the overridden method is also annotated with `@MustBeClosed`?

schlosna replied:
Yeah, that would help avoid some noise as this rolls out, then we can ratchet it up.

schlosna replied:
fixed by https://github.com/palantir/gradle-baseline/pull/1685

schlosna committed:
Apply AutoCloseableMustBeClosed

./gradlew compileJava compileTestJava -PerrorProneApply=AutoCloseableMustBeClosed
